### ISSUE TITLE
running on custom dataset

### ISSUE THREAD

**Issue by andreaceruti** [xiaozai/DeT#10]
Hi! I have already trained a detector on my dataset. What I would like to do now is to run a tracker of yours on my dataset, providing in input the results from detections. 
For example I have succesfully run some tracking algorithms as SORT, providing in input a format file as the [MOT standard input](https://motchallenge.net/instructions/).
can you guide me in how to provide my dataset in input to your method?

**Comment by xiaozai** [comment]
Hi, do you mean using your dataset in training or test?

**Comment by andreaceruti** [comment]
@xiaozai I would like to track grape bunches in video frames. What I am looking for is a tracking-by-detection framework that allows me to do the tracking step after my detector has provide the results. So to be clear I would like to do the tracking for each of these instances found in each frame as in this case. (At every frame I provide the detections and the tracker provide me the ID of each instance)
![rgb1934](https://user-images.githubusercontent.com/44469014/167503589-7a8251f3-f981-4213-b480-6aeb7e0db17d.png)

I am pretty new to object tracking and so I do not know if in my case I should do both train and test with your algorithm, sorry If I am not well prepared to answer your question

**Comment by xiaozai** [comment]
Hi, thanks for your interest! 
If you want to use your dataset in test, you need to 
   1) add a dataset.py under pytrtacking/evaluation/xxxdataset.py
   2) add the dataset path in pytracking/evaluation.local.py 
   3) add the dataset config in pytracking/evaluation/datasets.py 
   4) then run the command " python run_tracker.py tracker_name parameter_name --dataset_name your_dataset_name --sequence sequence_name --debug debug .... "

if you want to use your dataset in training, you need to :
   1) add dataset file under ltr/dataset/xxxdataset.py 
   2) add import in ltr/dataset/__init__.py 
   3) choose the training model in ltr/models/ 
   4) add your training settings in ltr/training_settings/tracker/xxx.py 
   5) run the command "python run_train.py tracker parameter"

If you only use the RGB data, I would like to recommend you to check the orginal Pytracking https://github.com/visionml/pytracking

**Comment by andreaceruti** [comment]
@xiaozai Sorry I have taken my time to read some papers and doing some researches.
In your paper there is written that:

> Our results verify that deep learning-based trackers, such as the proposed DeT, provide better accuracy than RGB trackers or ad hoc RGBD trackers on RGBD datasets, but they need to be pre-trained with real RGBD data (D is generated in our experiments) and fine-tuned using dataset specific training data.

So I should use DeT with my dataset as far as the categories I want to track are included in your training dataset. 
![Screenshot from 2022-05-15 16-32-01](https://user-images.githubusercontent.com/44469014/168478202-52a64cce-441a-4e4d-b02e-ad011f82fd9f.png)

My question is: since I want to track a category that does not belong to your trained categories, and also I want to use as input not only a video sequence, but a video sequence which already have the detections for each frame of my video, does it make sense to use your DeT ?

**Comment by xiaozai** [comment]
Hi, it seems that you want to perform tracking based on detection. You should try some detection based tracking methods. You can find the good examples here: https://github.com/yehengchen/Object-Detection-and-Tracking

Since DeT is the deep correlation filter based tracker, which use the DiMP and ATOM as base trackers. 
For DCF based trackers, they online trained a correlation filter of the tracked target (unknown target) based on the first frame. For detection based trackers, they perform tracking based on the detected instances, such as Yolo+SomeTrackers.
